### ISSUE TITLE
libcec: cec-client broken on arm64 / when HOMEBREW_PREFIX != /usr/local

### ISSUE THREAD

**Issue by micolous** [Homebrew/homebrew-core#127589]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
% brew info libcec
==> libcec: stable 6.0.2 (bottled)
Control devices with TV remote control and HDMI cabling
http://libcec.pulse-eight.com/
/opt/homebrew/Cellar/libcec/6.0.2 (39 files, 1MB) *
  Poured from bottle using the formulae.brew.sh API on 2023-04-05 at 17:03:55
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/libcec.rb
License: GPL-2.0
==> Dependencies
Build: cmake ✔
==> Analytics
install: 2 (30 days), 85 (90 days), 374 (365 days)
install-on-request: 2 (30 days), 85 (90 days), 374 (365 days)
build-error: 0 (30 days)

% brew config
HOMEBREW_VERSION: 4.0.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: 931327df1f30a3a5b1aa07657c0d1070e61988f9
Last commit: 5 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 6e79ffce447c18112ed09a075c199805d0a4a04d
Core tap last commit: 2 hours ago
Core tap branch: master
Core tap JSON: 05 Apr 06:35 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: /opt/homebrew/bin/joe
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_GOOGLE_ANALYTICS: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
Rosetta 2: false

% brew doctor
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Install `libcec`, then run `cec-client` on a machine where `HOMEBREW_PREFIX` is **not** `/usr/local`, such as `/opt/homebrew` (the default on Apple Silicon/ARM machines).

You should be able to run `cec-client -i` to print out information about how `libcec` was compiled, even if you don't have a [Pulse-Eight USB HDMI-CEC adapter][0] (which this software is a driver for).

This issue looks similar to https://github.com/Homebrew/brew/issues/13481.

It looks like `libcec` has a rather funky way of providing its API: instead of dynamically linking against `libcec.6.dylib` at compile time, the headers `dlopen("libcec.6.dylib")` at runtime, which uses the system library paths:

https://github.com/Pulse-Eight/libcec/blob/bf5a97d7673033ef6228c63109f6baf2bdbe1a0c/include/cecloader.h#L121-L122

```
% otool -L /opt/homebrew/Cellar/libcec/6.0.2/bin/cec-client
/opt/homebrew/Cellar/libcec/6.0.2/bin/cec-client:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.0.0)
	/usr/lib/libncurses.5.4.dylib (compatibility version 5.4.0, current version 5.4.0)
	/System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo (compatibility version 1.2.0, current version 1.5.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.32.0)
```

The `LibCecInitialise()` function allows a developer to pass in a custom path to `libcec.6.dylib`, but `cec-client` and the other binaries in this package never set that. As they're all packaged together, these should all "just work™" and that step should be unnecessary.

[0]: https://www.pulse-eight.com/p/104/usb-hdmi-cec-adapter

### What happened (include all command output)?

```
% brew install libcec
==> Downloading https://formulae.brew.sh/api/formula.jws.json
######################################################################## 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
######################################################################## 100.0%
==> Fetching libcec
==> Downloading https://ghcr.io/v2/homebrew/core/libcec/manifests/6.0.2
Already downloaded: /Users/michael/Library/Caches/Homebrew/downloads/338a07a42c6c56b6a7da37b616d42fbf36d8cb61c1a643a814e811ad1bab6828--libcec-6.0.2.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/libcec/blobs/sha256:8f1be1bc0bd
Already downloaded: /Users/michael/Library/Caches/Homebrew/downloads/44d2fd432a20884c919db6571b1a6337b2f95774fdcdef17b3da14e6c376d79a--libcec--6.0.2.arm64_ventura.bottle.tar.gz
==> Pouring libcec--6.0.2.arm64_ventura.bottle.tar.gz
🍺  /opt/homebrew/Cellar/libcec/6.0.2: 39 files, 1MB
==> Running `brew cleanup libcec`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).

% cec-client -i
dlopen(libcec.6.dylib, 0x0001): tried: '/libcec.6.dylib' (no such file), 'libcec.6.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibcec.6.dylib' (no such file), '/usr/lib/libcec.6.dylib' (no such file, not in dyld cache), 'libcec.6.dylib' (no such file), '/usr/local/lib/libcec.6.dylib' (no such file), '/usr/lib/libcec.6.dylib' (no such file, not in dyld cache)
```

### What did you expect to happen?

If you manually override the `DYLD_LIBRARY_PATH` environment variable, it works correctly:

```
% export DYLD_LIBRARY_PATH=/opt/homebrew/lib
% cec-client -i
libCEC version: 6.0.2, git revision: <unknown>, compiled on 2020-07-13 10:18:33 by brew@Ventura-arm64.local on Darwin 22.1.0 (arm64), features: P8_USB, DRM
```

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
# HOMEBREW_PREFIX != /usr/local
brew install libcec
cec-client -i
```

**Comment by carlocab** [comment]
Fixed in #127592.

**Comment by cho-m** [comment]
Closing as complete.

Let us know if there are still any issues after a `brew reinstall libcec`.
